h2o-assemblies/main/build.gradle

```diff
@@ -50,27 +50,39 @@ for (comp in optionalComponents) {
     
 }
 
-shadowJar {
-  mergeServiceFiles()
-  classifier = ''
-  // CDH 5.3.0 provides joda-time v1.6 which is too old, shadow the library instead
-  if (!project.hasProperty("jacocoCoverage")) {
-    relocate 'org.joda.time', 'ai.h2o.org.joda.time'
-  }
-  exclude 'META-INF/*.DSA'
-  exclude 'META-INF/*.SF'
-  exclude 'synchronize.properties'
-  exclude 'uploader.properties'
-  exclude 'test.properties'
-  exclude 'cockpitlite.properties'
-  exclude 'devpay_products.properties'
-  manifest {
-    attributes 'Main-Class': 'water.H2OApp'
-  }
+task shadowJar(type: ShadowJar) {
+    configurations = [ project.configurations.compile ]
+    mergeServiceFiles()
+    classifier = ''
+    // CDH 5.3.0 provides joda-time v1.6 which is too old, shadow the library instead
+    if (!project.hasProperty("jacocoCoverage")) {
+        relocate 'org.joda.time', 'ai.h2o.org.joda.time'
+    }
+    exclude 'META-INF/*.DSA'
+    exclude 'META-INF/*.SF'
+    exclude 'synchronize.properties'
+    exclude 'uploader.properties'
+    exclude 'test.properties'
+    exclude 'cockpitlite.properties'
+    exclude 'devpay_products.properties'
+    manifest {
+        attributes 'Main-Class': 'water.H2OApp'
+    }
+
+    into("META-INF") {
+        from("${buildDir.absolutePath}/reports/license"){
+            include "dependency-license.xml"
+        }
+    }
+}
+
+configurations{
```

Comment:
So that's why I need to create a new configuration and specify a custom `shadowJar` task.

The `shadowJar` provides `append` method, but that's for slightly different use-case